src/test/java/bio/terra/service/job/JobPermissionTest.java

```diff
@@ -172,28 +176,23 @@ public void testJobPermissions() throws Exception {
     assert (combinedIngestJobResponse.getStatusCode().is2xxSuccessful());
 
     // Verify custodian can view jobs
-    String datasetCreateJobId = jobResponse.getResponseObject().get().getId();
-    dataRepoFixtures.getJobSuccess(datasetCreateJobId, custodian());
+    JobModel datasetCreateJob = jobResponse.getResponseObject().get();
+    dataRepoFixtures.getJobSuccess(datasetCreateJob.getId(), custodian());
 
-    String fileIngestJobId = fileIngestJobResponse.getResponseObject().get().getId();
-    dataRepoFixtures.getJobSuccess(fileIngestJobId, custodian());
+    JobModel fileIngestJob = fileIngestJobResponse.getResponseObject().get();
+    dataRepoFixtures.getJobSuccess(fileIngestJob.getId(), custodian());
 
-    String bulkLoadJobId = bulkLoadJobResponse.getResponseObject().get().getId();
-    dataRepoFixtures.getJobSuccess(bulkLoadJobId, custodian());
+    JobModel bulkLoadJob = bulkLoadJobResponse.getResponseObject().get();
+    dataRepoFixtures.getJobSuccess(bulkLoadJob.getId(), custodian());
 
-    String metadataIngestJobId = metadataIngestJobResponse.getResponseObject().get().getId();
-    dataRepoFixtures.getJobSuccess(metadataIngestJobId, custodian());
+    JobModel metadataIngestJob = metadataIngestJobResponse.getResponseObject().get();
+    dataRepoFixtures.getJobSuccess(metadataIngestJob.getId(), custodian());
 
-    String combinedIngestJobId = combinedIngestJobResponse.getResponseObject().get().getId();
-    dataRepoFixtures.getJobSuccess(combinedIngestJobId, custodian());
+    JobModel combinedIngestJob = combinedIngestJobResponse.getResponseObject().get();
+    dataRepoFixtures.getJobSuccess(combinedIngestJob.getId(), custodian());
 
-    List<String> jobIds =
-        List.of(
-            datasetCreateJobId,
-            fileIngestJobId,
-            bulkLoadJobId,
-            metadataIngestJobId,
-            combinedIngestJobId);
+    List<JobModel> jobIds =
+        List.of(datasetCreateJob, fileIngestJob, bulkLoadJob, metadataIngestJob, combinedIngestJob);
 
     assertTrue(
```

Comment:
I noticed that the latest commit resulted in this integration test failing.  I looked at the failure briefly… what I noticed in the test output was that using `assertTrue` for a collection comparison results in a test that's harder to debug.

The additional logging is a definite improvement, but I found myself wanting a "you expected this collection but had this collection instead" kind of assertion instead.  Was this something you considered or experimented with when trying to fix this test?

Reply:
+1 for what @okotsopoulos said.
If I understand correctly, this error could have been caused by other factors, for example, if SAM didn't respond.  Is there a way to access the logs for these containers?  We might see errors for *hermione* around the time the action was executing.

Reply:
Yeah...remember what I was saying about preferring AssertJ?  It makes list comparisons like these a lot easier.  For now, I'm going to isolate this test to make sure that it doesn't run in parallel with others